packages/block-editor/src/components/block-inspector/index.js

```diff
@@ -314,6 +336,145 @@ const BlockInspectorSingleBlock = ( { clientId, blockName } ) => {
 	);
 };
 
+const NavigationInspector = ( { selectedBlockClientId, blockName } ) => {
+	const { parentNavBlock, childNavBlocks } = useSelect(
+		( select ) => {
+			const {
+				getBlockParentsByBlockName,
+				getClientIdsOfDescendants,
+				getBlock,
+			} = select( blockEditorStore );
+
+			let navBlockClientId;
+
+			if ( blockName === 'core/navigation' ) {
+				navBlockClientId = selectedBlockClientId;
+			} else if (
+				blockName === 'core/navigation-link' ||
+				blockName === 'core/navigation-submenu'
+			) {
+				navBlockClientId = getBlockParentsByBlockName(
+					selectedBlockClientId,
+					'core/navigation',
+					true
+				)[ 0 ];
+			}
+
+			const _childClientIds = getClientIdsOfDescendants( [
+				navBlockClientId,
+			] );
+
+			return {
+				parentNavBlock: getBlock( navBlockClientId ),
+				childNavBlocks: _childClientIds.map( ( id ) => {
+					return getBlock( id );
+				} ),
+			};
+		},
+		[ selectedBlockClientId, blockName ]
+	);
+
+	return (
+		<NavigatorProvider
+			initialPath={ selectedBlockClientId }
+			initialAnimationSettings={ { initial: false } }
+		>
+			<NavigationInspectorScreens
+				selectedBlockClientId={ selectedBlockClientId }
+				parentNavBlock={ parentNavBlock }
+				childNavBlocks={ childNavBlocks }
+			/>
+		</NavigatorProvider>
+	);
+};
+
+const NavigationInspectorScreens = ( {
+	selectedBlockClientId,
+	parentNavBlock,
+	childNavBlocks,
+} ) => {
+	const { goTo } = useNavigator();
+	const previousDepth = useRef( -1 );
+	const { navBlockTree } = useSelect(
+		( select ) => {
+			const { __unstableGetClientIdWithClientIdsTree } =
+				select( blockEditorStore );
+
+			return {
+				navBlockTree: __unstableGetClientIdWithClientIdsTree(
+					parentNavBlock.clientId
+				),
+			};
+		},
+		[ selectedBlockClientId ]
+	);
+
+	const getBlockDepth = ( targetClientId, currentDepth, rootBlock ) => {
+		if ( targetClientId === rootBlock.clientId ) {
+			return currentDepth;
+		}
+		for ( let i = 0; i < rootBlock.innerBlocks.length; i++ ) {
+			const newDepth = getBlockDepth(
+				targetClientId,
+				currentDepth + 1,
+				rootBlock.innerBlocks[ i ]
+			);
+			if ( newDepth > currentDepth ) {
+				return newDepth;
+			}
+		}
+	};
+
+	useEffect( () => {
+		const currentDepth = getBlockDepth(
+			selectedBlockClientId,
+			0,
+			navBlockTree
+		);
+		const animationSettings = {
+			initial: false,
+		};
+
+		if ( currentDepth === 0 && previousDepth.current > 0 ) {
+			animationSettings.initial = {
+				x: isRTL() ? 50 : -50,
+				opacity: 0,
+			};
+		} else if ( currentDepth > 0 && previousDepth.current === 0 ) {
+			animationSettings.initial = {
+				x: isRTL() ? -50 : 50,
+				opacity: 0,
+			};
+		}
+		previousDepth.current = currentDepth;
+		goTo( selectedBlockClientId, animationSettings );
```

Comment:
One of the points in favour of using the navigator component is that we won't have to duplicate the animation logic. If we're doing that here then that makes it a less compelling argument.